traits/testing/tests/test_optional_dependencies.py

```diff
@@ -0,0 +1,41 @@
+# -----------------------------------------------------------------------------
+#
+#  Copyright (c) 2019, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in /LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+# -----------------------------------------------------------------------------
+
+import sys
+from types import ModuleType
+import unittest
+
+from traits.testing.optional_dependencies import optional_import
+
+
+class MockModule(ModuleType):
+    """Mock module as object for testing"""
+    pass
+
+
+# Assign to sys modules to allow import
+sys.modules['mock_module'] = MockModule(name='mock')
```

Comment:
Do want to remove this from `sys.modules` after the test is complete? Maybe add it to `sys.modules` in the test setup and remove it in the teardown?

Reply:
Argh. Yes, definitely. I spotted this first time round and then forgot about it (and also forgot to comment on it).

Maybe it would be even better to use something that we know is importable, like one of the standard library modules.

Reply:
replacing the existing code with a stdlib module works for me.